File: Data Structures & Algorithms/majority-element/submission-1.java
class Solution {
    public int majorityElement(int[] nums) {
        int max_count = 0;
        int answer  = 0;
        Map<Integer, Integer> freq = new HashMap<>();

        for(int i : nums){
            int count = freq.getOrDefault(i, 0) + 1;
            freq.put(i, count);
            if(count > max_count){
                max_count = count;
                answer = i;
            }
        }
        return answer;
        
    }
}

File: Data Structures & Algorithms/maximum-subarray/submission-1.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        dp = [0] * len(nums)
        dp[0] = nums[0]

        for i in range(1, len(nums)):
            dp[i] = max(
                dp[i-1] + nums[i],
                nums[i]
            )
        return max(dp)
    
# 2  -3 4 -2 2 1 -1 4
# 2  -1 4  2 4 5  4  8